coreos-base/coreos-metadata/coreos-metadata-9999.ebuild

```diff
@@ -1,36 +1,222 @@
-# Copyright (c) 2015 CoreOS, Inc.. All rights reserved.
+# Copyright (c) 2017 CoreOS, Inc.. All rights reserved.
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-CROS_WORKON_PROJECT="coreos/coreos-metadata"
+EAPI=6
+
+CROS_WORKON_PROJECT="sdemos/coreos-metadata"
 CROS_WORKON_LOCALNAME="coreos-metadata"
 CROS_WORKON_REPO="git://github.com"
-COREOS_GO_PACKAGE="github.com/coreos/coreos-metadata"
-inherit coreos-go cros-workon systemd
 
 if [[ "${PV}" == 9999 ]]; then
 	KEYWORDS="~amd64 ~arm64"
 else
-	CROS_WORKON_COMMIT="8f2578dc5231e27f05e8d105d39c03c057946862" # v0.14.0
+	CROS_WORKON_COMMIT="76d22ec60d0bffa031d907f2c5818591f7134697" # v1.0.0
 	KEYWORDS="amd64 arm64"
 fi
 
-DESCRIPTION="A simple cloud-provider metadata agent"
-HOMEPAGE="https://github.com/coreos/coreos-metadata"
-SRC_URI=""
+inherit coreos-cargo cros-workon systemd
 
+DESCRIPTION="A tool for collecting instance metadata from various providers"
+HOMEPAGE="https://github.com/coreos/coreos-metadata"
 LICENSE="Apache-2.0"
 SLOT="0"
-IUSE=""
 
-src_compile() {
-	export GO15VENDOREXPERIMENT="1"
-	GO_LDFLAGS="-X main.version=$(git describe --dirty)" || die
-	go_build "${COREOS_GO_PACKAGE}/internal"
+src_unpack() {
+	cros-workon_src_unpack "$@"
+	coreos-cargo_src_unpack "$@"
+	unpack update-ssh-keys-0.1.0.tar.gz
```

Comment:
Let's make this version a variable defined at the top near the `CROS_` settings.  It appears enough times that updating it will be error prone.  So this line could be e.g. `unpack "update-ssh-keys-${UPDATE_SSH_KEYS_VERSION}.tar.gz"`.  A `grep || die` line could also be added to check that this variable actually matches what is in `Cargo.toml` to make sure required updates aren't missed.